bin/update_dev_data.py

```diff
@@ -42,21 +42,30 @@ def create_all(self):
         some fields, then overwrite those incorrect values with the standard
         values.
         """
-        for data_dict in self.devdata:
-            data_dict.pop("__doc__", None)
+        self.tm.begin()
+        for item in self.devdata:
+            getattr(self, f"upsert_{item['type']}", self.upsert_model)(**item)
 
-            type_ = data_dict.pop("type")
+        self.tm.commit()
 
-            if type_ == "application_instance":
-                self.upsert_application_instance(data_dict)
-            elif type_ == "assignment":
-                self.upsert_assignment(data_dict)
-            else:
-                raise RuntimeError(f"Unrecognized type: {type_}")
+    def upsert_model(self, **item):
+        data = item["data"]
 
-        self.tm.commit()
+        for key, value in data.items():
+            # Decode any base64 fields
+            if key.startswith("base64_"):
```

Comment:
Sort of generalized this base64 behavior.

Reply:
I like this, but I do wish it was a bit more obvious it's not part of the key. The keys here are `"base64_encoded_developer_secret_bytes` which really reads like:

 * `base64_encoded` / `developer_secret_bytes`

We could use some chars which can't be the attribute of python object like:

 * `base64:encoded_developer_secret_bytes`
 * `<base64>encoded_developer_secret_bytes`

I also wonder if value is the place to put it? The down side of this is whatever identifier you pick could be something that shows up in real data. For example we could use the [URL data format](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs) in the key for this, as it's somewhat standard:

`data:text/plain;base64,SGVsbG8sIFdvcmxkIQ==`